packages/date-picker/src/shared/components/DateInput/DateInputBox/DateInputBox.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { FocusEventHandler } from 'react';
+import { isSameDay } from 'date-fns';
+import isEqual from 'lodash/isEqual';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useForwardedRef } from '@leafygreen-ui/hooks';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+
+import {
+  DateSegmentsState,
+  isDateSegment,
+  useDateSegments,
+} from '../../../hooks';
+import {
+  doesSomeSegmentExist,
+  getValueFormatter,
+  newDateFromSegments,
+} from '../../../utils';
+import { useDatePickerContext } from '../../DatePickerContext';
+import { DateInputSegment } from '../DateInputSegment';
+import { DateInputSegmentChangeEventHandler } from '../DateInputSegment/DateInputSegment.types';
+
+import {
+  segmentPartsWrapperStyles,
+  separatorLiteralDisabledStyles,
+  separatorLiteralStyles,
+} from './DateInputBox.styles';
+import { DateInputBoxProps } from './DateInputBox.types';
+
+/**
+ * Renders a styled date input with appropriate segment order & separator characters.
+ *
+ * Depends on {@link DateInputSegment}
+ *
+ * Uses parameters `value` & `dateFormat` along with {@link Intl.DateTimeFormat.prototype.formatToParts}
+ * to determine the segment order and separator characters.
+ *
+ * Provided value is assumed to be UTC.
+ *
+ * Argument passed into `setValue` callback is also in UTC
+ * @internal
+ */
+export const DateInputBox = React.forwardRef<HTMLDivElement, DateInputBoxProps>(
+  (
+    {
+      value: dateValue,
```

Comment:
nit but these are internal only components - if you wanted to change the prop names so you didn't have to alias them, you could